sopel/modules/url.py

```diff
@@ -98,6 +98,14 @@ def setup(bot):
         bot.memory['shortened_urls'] = tools.SopelMemory()
 
 
+def shutdown(bot):
+    for key in ['url_exclude', 'last_seen_url', 'shortened_urls']:
+        try:
+            del bot.memory[key]
+        except KeyError:
+            pass
```

Comment:
Clearing `last_seen_url` makes sense, for the same reason that it makes sense to clear `find_lines` (the module could miss a URL while unloaded, and `.title` would then return possibly an unpredictable result).

Clearing `url_exclude` makes it possible to edit the exclude patterns and then reload the module for them to take effect, so that's good.

But clearing `shortened_urls` can only cause extra API calls, and I can't think of a way that leaving it behind on `shutdown()` could result in unexpected behavior.

Reply:
Cleared all but `shortened_urls`.